src/modules/vtol_att_control/standard.cpp

```diff
@@ -188,10 +188,10 @@ void Standard::update_vtol_state()
 
 					// The logic that will prevent spikes in airspeed measurements to trigger the transition
 					if ((_timestamp_transition_speed_detected == 0) &&
-					    (_airspeed_validated->calibrated_airspeed_m_s >= _params->transition_airspeed)) {
+					    (*_airspeed_smooth >= _params->transition_airspeed)) {
 						_timestamp_transition_speed_detected = hrt_absolute_time();
 
-					} else if (_airspeed_validated->calibrated_airspeed_m_s < _params->transition_airspeed) {
+					} else if (*_airspeed_smooth < _params->transition_airspeed) {
```

Comment:
We should consider some safeguard in case the smoothed airspeed is weird. How do we know that the smoothed airspeed actually is healthy and represents what we want? I see the param limits are conservative, that might be enough

Reply:
What additional safeguards do you think should be introduced after LPFing the signal?
Assessment was that this is redundant when the input topic is already validated, and would just increase code size.
At least if using a standard lpf implementation, shouldn't this be safe?

Reply:
For production implementation, the plan is to remove this entire code block, as an LPFed signal solved this issue by itself. Kept it in to remove change size. The slight risk of a transition timeout was considered acceptable for test flights.

Reply:
Good point. Since this is an exponential filter, we know that the true value will always be reached in approx 3 x VT_ASPD_TAU = 15s, regardless of the starting point. There are two failure modes:

1. Delayed triggering of forward/backward transition acceptance because of 3xVT_ASPD_TAU delay.
1a) Delayed forward transition acceptance by up to 15sec: Safeguarded by VT_TRANS_TIMEOUT.
2b) Delayed back transition acceptance by up to 15sec: Safeguarded by VT_B_TRANS_DUR like you say.

2. Premature trigger of forward/backward transition due to lagging VT_ASPD_TAU. Does VTOL attitude controller run during FW? Otherwise, the LPF might be at e.g. 19 mps at the end of front transition, and thus start out low when back-transition starts.
2a) Premature acceptance of forward transition: Altitude drop, adaptive QuadChute.
2b) Premature acceptance of back transition: Spontenaous "QuadChute"

In conclusion I agree that these risks seem acceptable, based on what you wrote. Next time it's super useful if the failure analysis is written down